hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HoodieHiveSyncClient.java

```diff
@@ -101,6 +101,9 @@ public HoodieHiveSyncClient(HiveSyncConfig config) {
           case JDBC:
             ddlExecutor = new JDBCExecutor(config);
             break;
+          case GLUE:
+            throw new HoodieHiveSyncException("GLUE mode is supported in AwsGlueCatalogSyncTool class only. "
```

Comment:
Instread of throwing inside the `HoodieHiveSyncClient`, I'm wondering why not just pick the right sync client in the caller.

Reply:
agreed. i can make that change :) . similar to -- https://github.com/apache/hudi/pull/11543.

lemme update this PR in combination instead.